File: go/src/auth/no_auth.go
package auth

import (
	"net/http"
)

// NoAuth implements the Authenticator interface for no authentication
type NoAuth struct {
	BaseAuth
}

// NewNoAuth creates a new NoAuth instance
func NewNoAuth() *NoAuth {
	return &NoAuth{
		BaseAuth: BaseAuth{
			Type: NoAuth,
		},
	}
}

// Authenticate always returns nil as no authentication is required
func (n *NoAuth) Authenticate(r *http.Request) error {
	return nil
} 

File: php/src/Auth/NoAuth.php
<?php

namespace Auth;

use Psr\Http\Message\ServerRequestInterface;

/**
 * NoAuth implementation that requires no authentication
 */
class NoAuth extends BaseAuth
{
    public function __construct()
    {
        $this->type = AuthType::NONE;
    }

    /**
     * Authenticate the request (always returns true as no auth is required)
     * @param ServerRequestInterface $request
     * @return bool
     */
    public function authenticate(ServerRequestInterface $request): bool
    {
        return true;
    }
} 